C error on non-existing struct field `chan` type

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.9, press to see full `v doctor` output</summary>

|V full version      |V 0.4.9 89d1aac.
|:-------------------|:-------------------
|OS                  |linux, "EndeavourOS Linux"
|Processor           |16 cpus, 64bit, little endian, AMD Ryzen 7 8845HS w/ Radeon 780M Graphics
|Memory              |21.77GB/30.64GB
|                    |
|V executable        |/home/lmp/Projects/v/v
|V last modified time|2025-03-19 09:21:17
|                    |
|V home dir          |OK, value: /home/lmp/Projects/v
|VMODULES            |OK, value: /home/lmp/.vmodules
|VTMP                |OK, value: /dev/shm
|Current working dir |OK, value: /home/lmp/Projects/v
|                    |
|Git version         |git version 2.49.0
|V git status        |weekly.2025.12-4-gb5283950 (2 commit(s) behind V master)
|.git/config present |true
|                    |
|cc version          |cc (GCC) 14.2.1 20250207
|gcc version         |gcc (GCC) 14.2.1 20250207
|clang version       |clang version 19.1.7
|tcc version         |tcc version 0.9.28rc 2024-07-31 HEAD@1cee0908 (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 0134e9b9
|emcc version        |emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 4.0.4-git (273f0216fede04f2445367765eaf2aabeeb60d84)
|glibc version       |ldd (GNU libc) 2.41

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  t.v && t`
```v
module main

struct ChanBug {
pub mut:
	bug chan DoesNotExist
}


```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: /dev/shm/t.01JPQN1HY8R0VJEKAF58A86TKR.tmp.c:2906: error: ';' expected (got "__chan_main__DoesNotExist_popval")
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

**What did you expect to see?**

A checker error (or other V error)


> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/ast/table.v
+++ b/vlib/v/ast/table.v
@@ -1499,9 +1499,13 @@ pub fn (t &Table) known_type_names() []string {
 	for _, idx in t.type_idxs {
 		typ := idx_to_type(idx)
 		// Skip `int_literal_type_idx` and `float_literal_type_idx` because they shouldn't be visible to the User.
-		if idx !in [0, int_literal_type_idx, float_literal_type_idx] && t.known_type_idx(typ)
-			&& t.sym(typ).kind != .function {
-			res << t.type_to_str(typ)
+		if idx !in [0, int_literal_type_idx, float_literal_type_idx] && t.known_type_idx(typ) {
+			tsym := t.sym(typ)
+			if tsym.kind !in [.function, .chan] {
+				res << t.type_to_str(typ)
+			} else if tsym.info is Chan && t.sym(tsym.info.elem_type).kind != .placeholder {
+				res << t.type_to_str(tsym.info.elem_type)
+			}
 		}
 	}
 	return res

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -5487,6 +5487,14 @@ fn (mut c Checker) ensure_type_exists(typ ast.Type, pos token.Pos) bool {
 				}
 			}
 		}
+		.chan {
+			if sym.info is ast.Chan {
+				if !c.ensure_type_exists((sym.info as ast.Chan).elem_type, token.Pos{ ...pos, col:
+					pos.col + 5 }) {
+					return false
+				}
+			}
+		}
 		else {}
 	}
 	return true

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -5489,8 +5489,10 @@ fn (mut c Checker) ensure_type_exists(typ ast.Type, pos token.Pos) bool {
 		}
 		.chan {
 			if sym.info is ast.Chan {
-				if !c.ensure_type_exists((sym.info as ast.Chan).elem_type, token.Pos{ ...pos, col:
-					pos.col + 5 }) {
+				if !c.ensure_type_exists((sym.info as ast.Chan).elem_type, token.Pos{
+					...pos
+					col: pos.col + 5
+				}) {
 					return false
 				}
 			}

--- a/vlib/v/checker/tests/chan_unknown_err.out
+++ b/vlib/v/checker/tests/chan_unknown_err.out
@@ -5,4 +5,3 @@ Did you mean `DoesNotExists`?
     7 |     bug chan DoesNotExist
       |              ~~~~~~~~~~~~
     8 | }
-    9 |

--- a/vlib/v/checker/tests/chan_unknown_err.out
+++ b/vlib/v/checker/tests/chan_unknown_err.out
@@ -1,0 +1,8 @@
+vlib/v/checker/tests/chan_unknown_err.vv:7:11: error: unknown type `DoesNotExist`.
+Did you mean `DoesNotExists`?
+    5 | struct ChanBug {
+    6 | pub mut:
+    7 |     bug chan DoesNotExist
+      |              ~~~~~~~~~~~~
+    8 | }
+    9 |

--- a/vlib/v/checker/tests/chan_unknown_err.vv
+++ b/vlib/v/checker/tests/chan_unknown_err.vv
@@ -1,0 +1,8 @@
+module main
+
+struct DoesNotExists {}
+
+struct ChanBug {
+pub mut:
+	bug chan DoesNotExist
+}

